source/common/upstream/upstream_impl.h

```diff
@@ -385,26 +382,24 @@ class ClusterInfoImpl : public ClusterInfo,
   const std::chrono::milliseconds connect_timeout_;
   absl::optional<std::chrono::milliseconds> idle_timeout_;
   const uint32_t per_connection_buffer_limit_bytes_;
-  Stats::ScopePtr stats_scope_;
-  mutable ClusterStats stats_;
   Stats::IsolatedStoreImpl load_report_stats_store_;
   mutable ClusterLoadReportStats load_report_stats_;
   Network::TransportSocketFactoryPtr transport_socket_factory_;
+  Stats::ScopePtr stats_scope_;
```

Comment:
Is stats_scope_ needed in this class?

Reply:
You'll need this to own the scope, it is passed by rvalue-ref.